.github/workflows/build-and-test.yml

```diff
@@ -487,6 +487,63 @@ jobs:
             throw "Failed to uninstall fluentdwinsvc."
           }
 
+  windows-zeroconfig-e2e-test:
+    runs-on: windows-2022
+    needs: [ windows-msi ]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache-dependency-path: '**/go.sum'
+
+      - name: Download Splunk OTel Collector msi
+        uses: actions/download-artifact@v3
+        with:
+          name: msi-build
+          path: ./tests/zeroconfig/windows/testdata/docker-setup/
+
+      - name: Copy install.ps1 to docker-setup folder
+        run: |
+          cp ./internal/buildscripts/packaging/installer/install.ps1 ./tests/zeroconfig/windows/testdata/docker-setup/
+
+      - name: Setup NuGet
+        uses: nuget/setup-nuget@v1.1.1
+
+      - name: Restore NuGet packages
+        working-directory: tests/zeroconfig/windows/testdata/apps/
+        run: |
+          nuget restore
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
+
+      - name: Build test application
+        working-directory: tests/zeroconfig/windows/testdata/apps/
+        run: |
+          msbuild .\AspNet.WebApi.NetFramework\AspNet.WebApi.NetFramework.csproj /p:Configuration=Release /p:Platform=AnyCPU /p:OutputPath=..\bin
+
+      - name: Create the firewall rule for the testutils OTLP sink
+        run: |
+          New-NetFirewallRule -DisplayName 'zc-iis-test' -Direction Inbound -LocalAddress 10.1.1.1 -LocalPort 4318 -Protocol TCP -Action Allow -Profile Any
+
+      - name: Build Docker compose
+        working-directory: tests/zeroconfig/windows/testdata/
+        run: |
+          docker compose build
+
+      - name: Run the test
+        working-directory: tests/zeroconfig/windows/
+        run: |
+          go test -timeout 5m -run ^TestWindowsIISInstrumentation$ -tags zeroconfig -v
```

Comment:
`-tags zeroconfig` could be used if added to the test file so we don't need to filter w/ `-run`